pypsdm/plots/grid.py

```diff
@@ -90,43 +185,176 @@ def grid_plot(
+        if cmap and colormap_value_dict and line_id in colormap_value_dict.keys():
+            value = colormap_value_dict[line_id]
+            colormap_value = _get_colormap_color(value, cmap)
+            use_colorbar = True
+        else:
+            colormap_value = "#008000"
+            use_colorbar = False
+
+    # Check for highlights (overrides colormap)
     if isinstance(highlights, dict):
-        for line_color, lines in highlights.items():
+        for color, lines in highlights.items():
             if line_data.name in lines:  # type: ignore
-                color = line_color
+                line_color = rgb_to_hex(color)
                 highlighted = True
+                use_colorbar = False
     elif highlights is not None:
         if line_data.name in highlights:
-            color = RED
+            line_color = rgb_to_hex(RED)
             highlighted = True
+            use_colorbar = False
 
+    # Handle disconnected lines
     if (highlight_disconnected is False) and is_disconnected:
         # Highlights override the disconnected status
         if not highlighted:
-            color = GREY
-
-    # Add the lines
-    fig.add_trace(
-        go.Scattermapbox(
-            mode="lines",
-            lon=lons,
-            lat=lats,
-            hoverinfo="skip",  # Skip hoverinfo for the lines
-            line=dict(color=rgb_to_hex(color)),
-        ),
-    )
+            line_color = rgb_to_hex(GREY)
+            use_colorbar = False
+
+    if cmap and colormap_value is not None:
+        hover_text += f"<br>{cbar_title or 'Value'}: {line_data_dict[line_id][next(iter(line_data_dict[line_id]))] * 100:.1f}%"
+
+    # Add the lines with or without colorbar
+    if colormap_value is not None:
+        if use_colorbar and show_colorbar is not None:
+            # Add line with colorbar support
+            fig.add_trace(
+                go.Scattermapbox(
+                    mode="lines",
+                    lon=lons,
+                    lat=lats,
+                    hoverinfo="skip",
+                    line=dict(color=colormap_value, width=2),
+                    showlegend=False,
+                )
+            )
+        else:
+            # Add regular line without colorbar
+            fig.add_trace(
+                go.Scattermapbox(
+                    mode="lines",
+                    lon=lons,
+                    lat=lats,
+                    hoverinfo="skip",  # Skip hoverinfo for the lines
+                    line=dict(color=line_color, width=2),
+                    showlegend=False,
+                )
+            )
+    else:
+        # Add regular line without colormap
+        fig.add_trace(
+            go.Scattermapbox(
+                mode="lines",
+                lon=lons,
+                lat=lats,
+                hoverinfo="skip",  # Skip hoverinfo for the lines
+                line=dict(color=line_color, width=2),
+                showlegend=False,
+            )
+        )
```

Comment:
This could be simplified. You could use if-else-statements to do `line_color = colormap_value` and have only one `fig._add_trace(...)` line.